doc/cookbook/source/examples/classifier/multiclass_ecoc_random.rst

```diff
@@ -13,9 +13,22 @@ There are multiple methods to encode or decode a codebook. In this example, we s
 Codebooks can also be encoded as dense or sparse. For dense codebooks, only :math:`+1` and :math:`-1` are generated as labels for each sample in each binary classifier. In
 sparse codebooks, :math:`+1`, :math:`-1` and :math:`0` are allowed, where :math:`0` labels the samples that are not classified.
 
-In this examples, we use :sgclass:`CECOCRandomDenseEncoder` as encoder. We also provides :sgclass:`CECOCRandomSparseEncoder` for generating sparse codebooks.
+In this examples, we use :sgclass:`CECOCRandomDenseEncoder` as encoder. :sgclass:`CECOCRandomSparseEncoder` can be applied for generating sparse codebooks.
 
-See :cite:`escalera2010ecoc` for a detailed introduction
+Shogun also provides:
+
+One-vs-Rest Encoder :sgclass:`CECOCOVREncoder`,
+One-vs-One Encoder :sgclass:`CECOCOVOEncoder`,
+Forest ECOC Encoder :sgclass:`CECOCForestEncoder` and
+Discriminant ECOC Encoder :sgclass:`CECOCDiscriminantEncoder`.
```

Comment:
I think you just want to reference the base class here. Not all the implementations